src/service/safety.ts

```diff
@@ -71,6 +73,29 @@ export class SafetyCalculator {
     private computeQtyLimit = () => {
         var settings = this._repo.latest;
 
+        var buyPing = 0;
+        var sellPong = 0;
+        var buyPq = 0;
+        var sellPq = 0;
+        var _buyPq = 0;
+        var _sellPq = 0;
+        for (var ti = this._broker._trades.length - 1; ti > -1; ti--) {
+          if (this._broker._trades[ti].side == Models.Side.Bid && buyPq<settings.size) {
```

Comment:
Took me a big effort to be able to get the list of the trades because im not expert of this language. Please feel free to optimize this because i can't make it better.

Thank you for your time''''